### PR TITLE
Devflow, boilerplate, improved code, formatting and bug fixes

### PR DESCRIPTION
**Pull request by kumare3**

### CODE REVIEW

**Comment by kumare3** [comment]
cc @evalsocket

**Review by evalsocket — COMMENTED** [review]
Looks good

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          fetch-depth: '0'
+      - name: Bump version and push tag
+        id: bump-version
+        uses: anothrNick/github-tag-action@1.17.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WITH_V: true
+          DEFAULT_BUMP: patch
+      - name: Push Docker Image to Github Registry
+        uses: whoan/docker-build-with-cache-action@v5
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply .github/workflows/master.yml:23]: Do we even need an image for flytectl?

[the remark this answers is not in the document]
Comment (evalsocket) [reply .github/workflows/master.yml:23]: No

File: Makefile [hunk 2]
```diff
@@ -1,2 +1,12 @@
+export REPOSITORY=flytectl
+include boilerplate/lyft/golang_test_targets/Makefile
+
 generate:
-	go test github.com/lyft/flytectl/cmd --update
\ No newline at end of file
+	go test github.com/lyft/flytectl/cmd --update
+
+compile:
```
Comment (wild-endeavor) [inline Makefile:7]: what's the difference between this? https://github.com/lyft/boilerplate/tree/master/boilerplate/lyft/flyte_golang_compile

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# WARNING: THIS FILE IS MANAGED IN THE 'BOILERPLATE' REPO AND COPIED TO OTHER REPOSITORIES.
```
Comment (wild-endeavor) [inline Dockerfile:1]: does this need a dockerfile?  if so, should we add this https://github.com/lyft/flyteadmin/blob/master/Makefile#L2 to the makefile which would add this https://github.com/lyft/boilerplate/blob/master/boilerplate/lyft/docker_build/Makefile

File: config.yaml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 admin:
+  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
   endpoint: http://localhost:30082
-  insecure: true
+  insecure: false
```
Comment (wild-endeavor) [inline config.yaml:5]: huh?

File: .github/workflows/master.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kumare3) [reply .github/workflows/master.yml:23]: we dont, this was just boilerplate, will remove it

[the remark this answers is not in the document]
Comment (kumare3) [reply .github/workflows/master.yml:23]: @evalsocket can you actually push the flytepropeller workflow to boilerplate, that way all repos can use it

File: Makefile (hunk 2 above)
Comment (kumare3) [reply Makefile:7]: not much, can change, i just added this for quick iteration

File: Dockerfile (hunk 3 above)
Comment (kumare3) [reply Dockerfile:1]: no dockerfile needed, removing

File: config.yaml (hunk 4 above)
Comment (kumare3) [reply config.yaml:5]: oops

File: .github/workflows/master.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (evalsocket) [reply .github/workflows/master.yml]: @kumare3  Are you talking about this https://github.com/lyft/boilerplate

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**